policy/release/lib/attestations.rego

```diff
@@ -134,3 +134,48 @@ task_succeeded(name) if {
 	task := task_in_pipelinerun(name)
 	task.status == "Succeeded"
 }
+
+_input_attestations := atts if {
+	# For ec validate image.
+	# We might not want this in the long term, but let's try to detect
+	# how we're being called. This is the "normal" situation when user
+	# runs `ec validate image` - the attestations were fetched already and
+	# are available in the input.
+	atts := input.attestations
+} else := atts if {
+	# For ec validate input.
+	# This is the new experimental part - extract the attestations at
+	# evaluation time
+	_default_sigstore_opts := {
+		"certificate_identity": "",
+		"certificate_identity_regexp": "",
+		"certificate_oidc_issuer": "",
+		"certificate_oidc_issuer_regexp": "",
+		"ignore_rekor": false,
+		"public_key": "",
+		"rekor_url": "",
+	}
+
+	# See hack/builtin-experiments/demo.sh in the ec-cli repo
+	_sigstore_opts := object.union(_default_sigstore_opts, data.sigstore_opts)
+
+	# Coerce into string since I think that is required
+	_image_ref := sprintf("%s", [input.image.ref])
+
+	# Get the attestations
+	info := ec.sigstore.verify_attestation(_image_ref, _sigstore_opts)
+
+	# This doesn't work for some reason.
+	# Todo: Fix this and add more checks here probably.
```

Comment:
Yeah, that's the challenge with this approach. It's hard to propagate errors. You have to explicitly look for them.